utils/src/main/scala/com/ibm/sparktc/sparkbench/utils/SparkFuncs.scala

```diff
@@ -19,23 +19,41 @@ package com.ibm.sparktc.sparkbench.utils
 
 import org.apache.spark.sql.functions.lit
 import org.apache.spark.sql.{DataFrame, SparkSession}
+import com.databricks.spark.avro._
 
 object SparkFuncs {
 
-  def writeToDisk(outputDir: String, data: DataFrame, spark: SparkSession, fileFormat: Option[String] = None): Unit = {
-
-    verifyPathNotExistsOrThrow(outputDir, s"Error: $outputDir already exists!", spark)
+  private def parseFormat(outputDir: String, fileFormat: Option[String]): String = (outputDir, fileFormat) match {
+    case (Formats.console, None) => Formats.console
+    case (_, None)         => outputDir.split('.').last
+    case (_, Some(s))      => s
+  }
 
-    val format = (outputDir, fileFormat) match {
-      case ("console", None) => "console"
-      case (_, None)         => outputDir.split('.').last
-      case (_, Some(s))      => s
+  def verifyCanWrite(outputDir: String, saveMode: String, spark: SparkSession, fileFormat: Option[String] = None): Boolean = {
+    if(outputDir == Formats.console) true
+    else {
+      (pathExists(outputDir, spark), saveMode) match {
+        case (false, _) => true
+        case (true, SaveModes.overwrite) => true
+        case (true, SaveModes.append) => true
+        case (true, SaveModes.error) => false
+        case (true, SaveModes.ignore) => true // allow write operation to no-op when the time comes
+        case (true, _) => throw SparkBenchException(s"Unrecognized or unimplemented save-mode: $saveMode")
+      }
     }
+  }
+
+  def writeToDisk(outputDir: String, saveMode: String, data: DataFrame, spark: SparkSession, fileFormat: Option[String] = None): Unit = {
+
+    val format = parseFormat(outputDir, fileFormat)
 
     format match {
-      case "parquet" => data.write.parquet(outputDir)
-      case "csv" => data.write.option("header", "true").csv(outputDir)
-      case "console" => data.show()
+      case Formats.parquet => data.write.mode(saveMode).parquet(outputDir)
+      case Formats.csv => data.write.mode(saveMode).option("header", "true").csv(outputDir)
+      case Formats.orc => data.write.mode(saveMode).orc(outputDir)
+      case Formats.avro => data.write.mode(saveMode).avro(outputDir)
+      case Formats.json => data.write.mode(saveMode).json(outputDir)
+      case Formats.console => data.show()
       case _ => throw new Exception(s"Unrecognized or unspecified save format. " +
         s"Please check the file extension or add a file format to your arguments: $outputDir")
```

Comment:
Can you move this check up to `verifyCanWrite` or something so that the format is checked before the workload run?  It's nice to error early.